### ISSUE TITLE
 Spotify only shows the Spotify's logo on a black background

### ISSUE THREAD

**Issue by Ralsa** [tiliado/nuvolaplayer#82]
Issue
-------

Spotify only shows the Spotify's logo on a black background

![](https://cloud.githubusercontent.com/assets/1562336/6855943/6dec795a-d3f4-11e4-8fee-9129f2158639.png)

Workaround
----------------

Until this issue is resolved, you have to remove Spotify's local storage when the issue occurs.

````
rm ~/.local/share/webkitgtk/localstorage/https_play.spotify.com_0.localstorage
```

Original Report
-------------------

LSB Version:	core-2.0-ia32:core-2.0-noarch:core-3.0-ia32:core-3.0-noarch:core-3.1-ia32:core-3.1-noarch:core-3.2-ia32:core-3.2-noarch:core-4.0-ia32:core-4.0-noarch:core-4.1-ia32:core-4.1-noarch:security-4.0-ia32:security-4.0-noarch:security-4.1-ia32:security-4.1-noarch
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.0 (jessie)
Release:	8.0
Codename:	jessie
Desktop: GNOME Shell 3.14.2

Nuvolaplayer version: 3.0.0+201503102245.603d6da
Revision: 603d6dafd0315b6d9c72
Spotify webapp integration script

 -Spotify screen only shows the icon in the upper right corner. Nothing more. The rest is black.
 -No records, neither hold messaging activity.

**Comment by fenryxo** [comment]
Hi. Thanks for taking the time to report this issue. I need more information to investigate it further:

 1. Are you behind HTTP/SOCKS proxy server?
 2. Are you behind a firewall?
 3. Is the issue reproducible in WebKitGTK MiniBrowser? Run  `/usr/lib/x86_64-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser https://play.spotify.com` or `/usr/lib/i386-linux-gnu/webkit2gtk-3.0/libexec/MiniBrowser https://play.spotify.com` in terminal.
 4. Does the issue persist with disabled DRI3? Run `export LIBGL_DRI3_DISABLE=1; nuvolaplayer3 -D` in terminal.

**Comment by ItzBlitz98** [comment]
I just ran into the exact same issue, 
If you go to: 

    ~/.local/share/webkitgtk/localstorage

and delete 

    https_play.spotify.com_0.localstorage


That should do the trick.

**Comment by Ralsa** [comment]
The trick works ok. 
Thank you very much.



El Jueves 19 marzo 14:41:31 Itzblitz98 escribió:


I just ran into the exact same issue, If you go to: 
~/.local/share/webkitgtk/localstorage
 
and delete 
https_play.spotify.com_0.localstorage
 
That should do the trick. 
—Reply to this email directly or view it on GitHub[1].[2]




--------
[1] https://github.com/tiliado/nuvolaplayer/issues/82#issuecomment-83770000
[2] 
https://github.com/notifications/beacon/AK9GoE4RnTCF508PgXhSkTqHfdYVE5xvks5n2zoL
gaJpZM4Du7Ys.gif

**Comment by fenryxo** [comment]
It might be interesting to find out whether the issue is in WebKit and how to avoid it. If anyone have the same issue in the future, don't delete (or backup) the localstorage file for further investigation. Thanks!

**Comment by ItzBlitz98** [comment]
Huh well its just happened again.

Here is the file i normally delete:
````
https://www.dropbox.com/s/p322xkrr03v57hx/https_play.spotify.com_0.localstorage?dl=0
````

Spotify does return a console error but its pretty meaningless:
![screenshot from 2015-03-26 20-12-01](https://cloud.githubusercontent.com/assets/1562336/6855943/6dec795a-d3f4-11e4-8fee-9129f2158639.png)

Will try the commit you just submitted and report back if any errors arise.

**Comment by fenryxo** [comment]
> Will try the commit you just submitted and report back if any errors arise.

The change in the commit actually works only if you have WebKitGTK+ >= 2.8.

It might help to disable local storage feature for Spotify. Will take a look at it in April as funds for March are almost all spent.

**Comment by fenryxo** [comment]
This ticket has been added to the roadmap for July 2015 tiliado/nuvolaplayer.project#4

**Comment by fenryxo** [comment]
Fix ready for review from the maintainer: tiliado/nuvola-app-spotify#5

**Comment by fenryxo** [comment]
Btw, thanks, @ItzBlitz98, for the broken localstorage file. It was very helpful.

**Comment by ItzBlitz98** [comment]
No problem thank you very much for looking into this the fix is much appreciated.

**Comment by fenryxo** [comment]
Released in Spotify script 2.1.
